Final fixes in extension code

abhishekgupta5 opened the pull request:
**Issue:** https://linear.app/nebulab-retainers/issue/ABU-26/fixes-in-extension

**Brief:** 
1. The variables in JS file should be prefixed using pt since they are easy to clash with app variables
2. BE specific issues: 
    a. The JS file is not loading when extension is installed. This is because we are adding `spree/backend/solidus_pay_tomorrow` in install_generator but it doesn't exist in extension. Hence removed it

    b. `app/views/spree/admin/payments/source_forms/_pay_tomorrow.html.erb` file should exist. Otherwise code breaks if we try to create a PT payment from BE.

3. Make the SolidusPayTomorrow::PaymentMethod creation idempotent in seed file. Everytime we run the generator, a new payment method is created.


- [x] QAed paytomorrow payments from frontend in sandbox and in Abunda

DanielePalombo commented on app/assets/javascripts/spree/backend/solidus_pay_tomorrow.js:
```diff
@@ -1,2 +1,4 @@
\ No newline at end of file
+// the installer will append this file to the app vendored assets here: vendor/assets/javascripts/spree/backend/all.js'
+
+//= require solidus_pay_tomorrow
```
I tested it in the backend and PayTomorrow gives me an error when I visit it.

http://localhost:3000/admin/orders/R104148028/payments/new

Can you confirm you are getting the same error?

abhishekgupta5 replied:
Yes, getting the same error. Fixed in https://github.com/nebulab/solidus_pay_tomorrow/pull/19/commits/e768d8b347b0580bca06d5f76f0956d5195033a0

Since URL's aren't working, I have not been able to QA a successful checkout. After success/cancel we are redirected to `checkout_state_path('confirm')` or `checkout_state_path('payment')` so this might break.

DanielePalombo replied:
What do you think about just disabling it in the backend? 
Maybe we can just show a message like:

`PayTomorrow can be performed only on frontend`

abhishekgupta5 replied:
Yea, I am thinking the same.

abhishekgupta5 replied:
Done in https://github.com/nebulab/solidus_pay_tomorrow/pull/19/commits/494c26110345801725467d2acb3d14bcbe661cc6

DanielePalombo commented on app/views/spree/shared/_pay_tomorrow.html.erb:
```diff
@@ -0,0 +1,13 @@
+
+<div
+  id="pay-tomorrow-order-payload"
+  data-frontend=<%= frontend %>
```
What is the purpose of this option?

abhishekgupta5 replied:
Removed this since we are not doing BE

abhishekgupta5 committed:
Handle BE specific code

1. Added the partial that'll show that PayTomorrow payments
   aren't available on BE.
   This is because the redirect on OrderApplicationController is done
   by PayTomorrow and we don't have info on order. We'll find a clean
   way to handle this later. Until then, we'll show
   "PayTomorrow payments are only available on FE right now"
2. Removed the spree/frontend/solidus_pay_tomorrow from
   install_generator

abhishekgupta5 committed:
Renamed pay-tomorrow specific variables

orderNumber and paymentMethodId can already be defined in the app
using the extension and hence it gives "Redeclaration of variable"
warning. Hence made these variables PayTomorrow specific

abhishekgupta5 committed:
Make the pay_tomorrow seed rake task idempotent

Did this to avoid creating multiple payment methods
if we run install generator multiple times
